sdk/data/azcosmos/cosmos_global_endpoint_manager_test.go

```diff
@@ -0,0 +1,378 @@
+	assert.NoError(t, err)
+
+	// Get the account properties
+	accountProps, err := gem.GetAccountProperties()
+	assert.NoError(t, err)
+
+	// Assert the expected account properties
+	expectedAccountProps := accountProperties{
+		ReadRegions:                  []accountRegion{},
+		WriteRegions:                 []accountRegion{},
+		EnableMultipleWriteLocations: false,
+	}
+	assert.Equal(t, expectedAccountProps, accountProps)
+}
+
+func TestGlobalEndpointManager_GetLocation(t *testing.T) {
+	// Create a mock client
+	headerPolicy := &headerPolicies{}
+	srv, close := mock.NewTLSServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	verifier := headerPoliciesVerify{}
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{PerCall: []policy.Policy{headerPolicy, &verifier}}, &policy.ClientOptions{Transport: srv})
+	req, err := azruntime.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		return
+	}
+	req.SetOperationValue(pipelineRequestOptions{
+		isWriteOperation: true,
+	})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+	mockClient := client
+
+	// Create a new global endpoint manager with preferred regions and refresh interval
+	gem, err := newGlobalEndpointManager(mockClient, []string{"westus", "eastus"}, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Get the location for the given endpoint
+	endpoint := url.URL{Scheme: "https", Host: "westus.documents.azure.com"}
+	location := gem.GetLocation(endpoint)
+
+	// Assert the expected location
+	expectedLocation := "westus"
+	assert.Equal(t, expectedLocation, location)
+}
+
+func TestGlobalEndpointManager_MarkEndpointUnavailableForRead(t *testing.T) {
+	// Create a mock client
+	headerPolicy := &headerPolicies{}
+	srv, close := mock.NewTLSServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	verifier := headerPoliciesVerify{}
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{PerCall: []policy.Policy{headerPolicy, &verifier}}, &policy.ClientOptions{Transport: srv})
+	req, err := azruntime.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		return
+	}
+	req.SetOperationValue(pipelineRequestOptions{
+		isWriteOperation: true,
+	})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+	mockClient := client
+
+	// Create a new global endpoint manager with preferred regions and refresh interval
+	gem, err := newGlobalEndpointManager(mockClient, []string{"westus", "eastus"}, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Mark an endpoint as unavailable for read
+	endpoint, err := url.Parse(client.endpoint)
+	if err != nil {
+		return
+	}
+
+	err = gem.MarkEndpointUnavailableForRead(*endpoint)
+	assert.NoError(t, err)
+
+	// Check if the endpoint is marked as unavailable for read
+	unavailable := gem.IsEndpointUnavailable(*endpoint, 1)
+	assert.True(t, unavailable)
```

Comment:
Also validate that unavailable region is excluded from readEndpoints afterwards.

Reply:
Same for below TestGlobalEndpointManager_MarkEndpointUnavailableForWrite test